frame.go

```diff
@@ -1452,49 +1452,66 @@ func (f *framer) writeQueryParams(opts *queryParams) {
 		return
 	}
 
-	var flags byte
-	if len(opts.values) > 0 {
-		flags |= flagValues
-	}
-	if opts.skipMeta {
-		flags |= flagSkipMetaData
-	}
-	if opts.pageSize > 0 {
-		flags |= flagPageSize
-	}
-	if len(opts.pagingState) > 0 {
-		flags |= flagWithPagingState
-	}
-	if opts.serialConsistency > 0 {
-		flags |= flagWithSerialConsistency
-	}
-
+	var flagsByte byte
+	var flagsInt uint32
 	names := false
 
-	// protoV3 specific things
-	if f.proto > protoVersion2 {
+	switch {
+	case f.proto < protoVersion5:
+		if len(opts.values) > 0 {
+			flagsByte |= byte(flagValues)
+		}
+		if opts.skipMeta {
+			flagsByte |= byte(flagSkipMetaData)
+		}
+		if opts.pageSize > 0 {
+			flagsByte |= byte(flagPageSize)
+		}
+		if len(opts.pagingState) > 0 {
+			flagsByte |= byte(flagPageSize)
+		}
+		if opts.serialConsistency > 0 {
+			flagsByte |= byte(flagWithSerialConsistency)
+		}
+		if f.proto > protoVersion2 {
+			if opts.defaultTimestamp {
+				flagsByte |= byte(flagDefaultTimestamp)
+			}
+			if len(opts.values) > 0 && opts.values[0].name != "" {
+				flagsByte |= byte(flagWithNameValues)
+				names = true
+			}
+		}
+		f.writeByte(flagsByte)
+
+	case f.proto >= protoVersion5:
```

Comment:
I think we also may avoid these code duplications by implementing only the new way of constructing the flag. As I see, we may construct the flag as **uint32**, and in the end, we may check the proto version. If the version is less than 5 we may cast it to **byte** and write it as it is.
What do you think about this approach? Also, if we follow this approach, the `flagsByte` var may be omitted.

Reply:
Make sense, fixed